power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -685,11 +701,31 @@ class Deserializer {
     }
 
     void parse_component(Idx component_idx) {
+        if (dataset_handler_.is_row_based(component_idx)) {
+            parse_component(row_based, component_idx);
+        } else {
+            parse_component(columnar, component_idx);
+        }
```

Comment:
I found out that this call will cause major overhead when no attributes are present. That means that the logic concerning empty attribute buffers is incorrect

```cpp
        } else if (!dataset_handler_.get_buffer(component_idx).attributes.empty()) {
            parse_component(columnar, component_idx);
        }
```

This is not a sustainable fix (it points at the fact that there is some other problem) but will resolve the issue in the short-term

| data | serializer creation | serialization | serialization total | deserializer creation + pre-parse | deserializer parse | deserializer total |
| -- | -- | -- | -- | -- | -- |-- |
msgpack [compact] | 0.0004923000233247876 s | 2.3669713999843225 s | 3.0552195999771357 s | 0.28873490006662905 s | 2.858401000034064 s | 3.285264000063762 s
msgpack [regular] | 0.0007754000835120678 s | 2.9432286999654025 s | 3.9057689999463037 s | 0.5173835000023246 s | 5.383461399935186 s | 6.166236900025979 s
json [compact] | 0.00043050001841038465 s | 29.926429499988444 s | 63.13568380009383 s | 14.445208700024523 s | 3.2074557000305504 s | 19.59107560000848 s
json [regular] | 0.0008066999725997448 s | 37.490393099957146 s | 78.83358660002705 s | 18.796793300076388 s | 5.285955099971034 s | 26.85202120000031 s

Reply:
What has changed in your suggestion to avoid the overhead?

Reply:
Your test scripts are testing row-based de-serialization right? So the `else` or `else if` in the segment is never reached.

Reply:
turns out it is... there's definitely an issue. I have very little time left before i need to go, though. Should we roll-back?

Reply:
ok found the problem in my test script (`auto` where there should've been `auto&` when declaring the dataset)

Reply:
I see. So, the buffers are never set in the de-serializer. And all the components are treated column-based  with no attributes.